fit1d.py

```diff
@@ -0,0 +1,328 @@
+#
+# This module reproduces much of the behaviour of IRAF 1-d fitting routines
+
+import math
+import numpy as np
+from scipy.interpolate import LSQUnivariateSpline
+
+def fit1d(x, y, weights=None,
+          naverage=1,
+          function="spline3",
+          order=3,
+          low_reject=3.0,
+          high_reject=3.0,
+          niterate=0,
+          grow=0.0):
+    """Main 1-d fitting function.  The parameters can be used exactly like they are in IRAF tasks
+    like fit1d, response, etc.  The default behaviour is to fit a cubic spline with order 3
+    (3 equal regions, 2 knots) with no sigma clipping.
+
+    Parameters
+    ----------
+
+    x : 1-d ndarray
+        The array of independent values
+
+    y : 1-d ndarray
+        The array of dependent values.  Must have same length as x, or function returns None
+
+    weights : 1-d ndarray or None
+        The array of weights.  If set to None, the points are given equal weight of 1.0.  If not
+        set to None, must have the same length as x and y, or the function returns None
+
+    naverage : int
+        The number of adjacent elements of x, y, and weight that are averaged before
+        fitting is done.  If naverage=1, no averaging is done.
+
+    function : str
+        Fitting function.  Currently only "spline1" and "spline3" are supported
+
+    order : int
+        Order of fitting fuction.  The number of knots in the spline fit is (order-1)
```

Comment:
The only thing I would add to previous comments is that this description of `order` should be more like the description in `fit_once`.